integration_tests/secretstorage_acceptance_test.go

```diff
@@ -0,0 +1,172 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package integrationtests
+
+import (
+	"context"
+	"errors"
+	"math/rand"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/redhat-appstudio/remote-secret/pkg/secretstorage"
+	"github.com/redhat-appstudio/remote-secret/pkg/secretstorage/awsstorage/awscli"
+	"github.com/redhat-appstudio/remote-secret/pkg/secretstorage/memorystorage"
+	"github.com/redhat-appstudio/remote-secret/pkg/secretstorage/vaultstorage"
+
+	"github.com/stretchr/testify/assert"
+	"k8s.io/apimachinery/pkg/util/uuid"
+)
+
+// Testsuite that runs same tests against multiple tokenstorage implementations.
+// Tests cares just about TokenStorage interface and makes sure that implementations behaves the same.
+// Runs against real storages, see comments on Test* functions for more details.
+
+// TestVaultStorage runs against local in-memory Vault. No external dependency or setup is needed, test runs everytime.
+func TestVaultStorage(t *testing.T) {
+	cluster, storage := vaultstorage.CreateTestVaultSecretStorage(t)
+
+	ctx := context.TODO()
+	assert.NoError(t, storage.Initialize(ctx))
+	defer cluster.Cleanup()
+
+	testStorage(t, ctx, storage)
+}
+
+// TestInMemoryStorage runs against our testing in-memory implementation of the TokenStorage.
+func TestInMemoryStorage(t *testing.T) {
+	storage := &memorystorage.MemoryStorage{}
+
+	ctx := context.TODO()
+	assert.NoError(t, storage.Initialize(ctx))
+
+	testStorage(t, ctx, storage)
+}
+
+// TestAws runs against real AWS secret manager.
+// AWS_CONFIG_FILE and AWS_CREDENTIALS_FILE must be set and point to real files with real credentials for testsuite to properly run. Otherwise test is skipped.
+func TestAws(t *testing.T) {
+	ctx := context.TODO()
+
+	awsConfig, hasAwsConfig := os.LookupEnv("AWS_CONFIG_FILE")
+	awsCreds, hasAwsCreds := os.LookupEnv("AWS_CREDENTIALS_FILE")
+
+	if !hasAwsConfig || !hasAwsCreds {
+		t.Log("AWS storage tests didn't run!. to test AWS storage, set AWS_CONFIG_FILE and AWS_CREDENTIALS_FILE env vars")
+		return
+	}
+
+	if _, err := os.Stat(awsConfig); errors.Is(err, os.ErrNotExist) {
+		t.Logf("AWS storage tests didn't run!. AWS_CONFIG_FILE is set, but file does not exists '%s'\n", awsConfig)
+		return
+	}
+
+	if _, err := os.Stat(awsCreds); errors.Is(err, os.ErrNotExist) {
+		t.Logf("AWS storage tests didn't run!. AWS_CREDENTIALS_FILE is set, but file does not exists '%s'\n", awsCreds)
+		return
+	}
+
+	secretStorage, error := awscli.NewAwsSecretStorage(ctx, "spi-test", &awscli.AWSCliArgs{
+		ConfigFile:      awsConfig,
+		CredentialsFile: awsCreds,
+	})
+	assert.NoError(t, error)
+	assert.NotNil(t, secretStorage)
+
+	err := secretStorage.Initialize(ctx)
+	assert.NoError(t, err)
+
+	testStorage(t, ctx, secretStorage)
+}
+
+func testStorage(t *testing.T, ctx context.Context, storage secretstorage.SecretStorage) {
+	refreshTestData(t)
+
+	// get non-existing
```

Comment:
Maybe wrap the individual variants in `t.Run("get non-existing", func...)` so that the failure reporting is better?